### ISSUE TITLE
Unbounded month calculations can exceed limits.

### ISSUE THREAD

**Issue by germinator** [kizitonwose/CalendarView#91]
Possibly bad luck on timing, but if you look at the PR I've provided, you'll see that dayValue can sometimes end up with a value of "32" or other values that are outside of the range of days of a month. (I believe I found this because of my example in February, which only has 29 days.)

I've changed the way days get incremented and added by using "plusDays" which will roll from month to month where required.

PR #90

**Comment by kizitonwose** [comment]
I've looked at the PR and it won't work for the second row of outdates. After 1-7, we need to continue from 8 upwards in the next loop, the logic in the PR does not accommodate this. I'd like to find out why the current logic is failing. Can you give me your calendar setup so I can try to reproduce the bug you have in February?

**Comment by germinator** [comment]
Thank you for looking at it.

There is a new test in the PR, in `CalendaerViewTests.kt`, that will fail if you revert the line in `MonthConfig.kt`. That will reproduce the problem by calling `MonthConfig.generateUnboundedMonths(...)` with the same parameters as I saw in our app. (Or at least it does for me.)

Does that work for you?

**Comment by kizitonwose** [comment]
Okay, I found the fix. Replace lines 156 - 169 in `MonthConfig` class with:

```kotlin
val nextRowDates = (1..7).map {
    CalendarDay(lastDay.date.plusDays(it.toLong()), DayOwner.NEXT_MONTH)
}
```

Then add some comments to the new test so it's clear what's going on, especially since there are no assertions. It should be good to merge.

**Comment by germinator** [comment]
Updated. Thank you for your feedback.

**Comment by kizitonwose** [comment]
The fix is now available in version `0.3.0`. Thank you for helping out!
